src/context.rs

```diff
@@ -66,6 +69,22 @@ pub struct SessionContext {
     pub remote_pubkey: Option<PublicKey>,
     /// Session is closed
     pub closed: Arc<AtomicBool>,
+    /// Session pending data size
+    pub pending_data_size: Arc<AtomicUsize>,
+}
+
+impl SessionContext {
+    // Increase when data pushed to Service's write buffer
+    pub(crate) fn incr_pending_data_size(&self, data_size: usize) {
+        self.pending_data_size
+            .fetch_add(data_size, Ordering::SeqCst);
```

Comment:
I think `Relaxed` is enough.

Reply:
You are right

Reply:
But If we want to use the counter later to do flood control,  maybe using `Acquire` in `incr`, and `Release` in `decr` is better.

Reply:
I think unless profile show this is a bottleneck otherwise no need to optimize for this. In flood control case final consistency is enough.